### ISSUE TITLE
 I would like to go through my wildcard in order without others having this function

### ISSUE THREAD

**Issue by Hezow** [adieyal/sd-dynamic-prompts#617]
In most of my prompts, I have several wildcards but I would like one of them to perfectly follow the order in which I listed my terms.

Thanks again for this extension, it's the most useful.

**Comment by rjgoif** [comment]
If I understand you correctly, you mean "I would like to specify some wildcards as combinatorial, and others as random", correct?

If so, I would also find this feature extremely useful.

**Comment by Hezow** [comment]
Yes, that's pretty much it, I had several wildcards and by default the terms selected in them are completely random. And I wanted one of them not to be in these conditions, I wanted it to follow the order in which I made my list. 

I found a solution with the Jinja2 Template. If you want I'll share it with you.

**Comment by anothertal3** [comment]
> I found a solution with the Jinja2 Template. If you want I'll share it with you.

Could be useful for me and maybe some others. Would you be so kind as to share your solution? Thank you.

**Comment by akx** [comment]
You should be able to use the variant/wildcard prefix option (which, apparently, hasn't really been documented!), implemented in https://github.com/adieyal/dynamicprompts/pull/27

If your default mode is combinatorial, you can do `__~foo__` to force random sampling for the `foo` wildcard (or `{~A|B|C}` for a variant).

Similarly, in random mode you can use `@` to force a given command to be interpreted in combinatorial mode.

**Comment by adieyal** [comment]
It has been documented, but perhaps not easy to find - https://github.com/adieyal/sd-dynamic-prompts/blob/main/docs/SYNTAX.md#cyclical-sampler
